File: token-client-core/src/test/java/no/nav/security/token/support/client/core/ClientAuthenticationPropertiesTest.java
package no.nav.security.token.support.client.core;


import com.nimbusds.oauth2.sdk.auth.ClientAuthenticationMethod;
import org.junit.jupiter.api.Test;

import static com.nimbusds.oauth2.sdk.auth.ClientAuthenticationMethod.NONE;
import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
import static org.junit.jupiter.api.Assertions.assertNotNull;

class ClientAuthenticationPropertiesTest {

    @Test
    void invalidAuthenticationProperties() {
        assertThatExceptionOfType(IllegalArgumentException.class)
            .isThrownBy(() -> instanceWith(ClientAuthenticationMethod.TLS_CLIENT_AUTH));
        assertThatExceptionOfType(IllegalArgumentException.class)
            .isThrownBy(() -> instanceWith(ClientAuthenticationMethod.SELF_SIGNED_TLS_CLIENT_AUTH));
        assertThatExceptionOfType(IllegalArgumentException.class)
            .isThrownBy(() -> instanceWith(ClientAuthenticationMethod.CLIENT_SECRET_JWT));
        assertThatExceptionOfType(IllegalArgumentException.class)
            .isThrownBy(() -> instanceWith(NONE));

        assertThatExceptionOfType(IllegalArgumentException.class)
            .isThrownBy(() -> ClientAuthenticationProperties.builder("client1", NONE)
                .build());
    }

    private static void instanceWith(ClientAuthenticationMethod clientAuthenticationMethod) {
        new ClientAuthenticationProperties(
            "client",
            clientAuthenticationMethod,
            "secret",
            null);
    }

}

File: token-client-core/src/test/kotlin/no/nav/security/token/support/client/core/ClientAuthenticationPropertiesTest.kt
package no.nav.security.token.support.client.core

import com.nimbusds.oauth2.sdk.auth.ClientAuthenticationMethod
import com.nimbusds.oauth2.sdk.auth.ClientAuthenticationMethod.CLIENT_SECRET_JWT
import com.nimbusds.oauth2.sdk.auth.ClientAuthenticationMethod.NONE
import com.nimbusds.oauth2.sdk.auth.ClientAuthenticationMethod.SELF_SIGNED_TLS_CLIENT_AUTH
import com.nimbusds.oauth2.sdk.auth.ClientAuthenticationMethod.TLS_CLIENT_AUTH
import no.nav.security.token.support.client.core.ClientAuthenticationProperties.Companion.builder
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows

internal class ClientAuthenticationPropertiesTest {

    @Test
    fun invalidAuthenticationProperties() {
        assertThrows<IllegalArgumentException> { instanceWith(TLS_CLIENT_AUTH) }
        assertThrows<IllegalArgumentException> { instanceWith(SELF_SIGNED_TLS_CLIENT_AUTH) }
        assertThrows<IllegalArgumentException> { instanceWith(CLIENT_SECRET_JWT) }
        assertThrows<IllegalArgumentException> { instanceWith(NONE) }
        assertThrows<IllegalArgumentException> {  builder("client1", NONE).build() }
    }

    private fun instanceWith(clientAuthenticationMethod : ClientAuthenticationMethod) =
        ClientAuthenticationProperties("client", clientAuthenticationMethod, "secret",
            null)

}